File: backend/app/api/v1/exercises.py
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.auth import get_current_telegram_id
from app.core.database import get_session

router = APIRouter()


@router.get("/muscle-groups")
async def get_muscle_groups(
    telegram_id: int = Depends(get_current_telegram_id),
    session: AsyncSession = Depends(get_session),
):
    """Return all muscle groups."""
    # TODO: implement via ExerciseService
    pass


@router.get("/muscle-groups/{group_id}")
async def get_exercises_by_group(
    group_id: int,
    telegram_id: int = Depends(get_current_telegram_id),
    session: AsyncSession = Depends(get_session),
):
    """Return all exercises for a muscle group."""
    pass


@router.get("/{exercise_id}")
async def get_exercise(
    exercise_id: int,
    telegram_id: int = Depends(get_current_telegram_id),
    session: AsyncSession = Depends(get_session),
):
    """Return a single exercise card."""
    pass

File: frontend/src/api/exercises.ts
import apiClient from './client'

export interface MuscleGroup {
  id: number
  name: string
}

export interface Exercise {
  id: number
  name: string
  muscle_group: string
  muscle_group_id: number
  description: string
  video_file_id: string | null
}

export async function getMuscleGroups(): Promise<MuscleGroup[]> {
  const res = await apiClient.get<MuscleGroup[]>('/exercises/muscle-groups')
  return res.data
}

export async function getExercisesByGroup(groupId: number | string): Promise<Exercise[]> {
  const res = await apiClient.get<Exercise[]>(`/exercises/muscle-groups/${groupId}`)
  return res.data
}

export async function getExerciseById(exerciseId: number | string): Promise<Exercise> {
  const res = await apiClient.get<Exercise>(`/exercises/${exerciseId}`)
  return res.data
}
